vscode/src/extension.ts

```diff
@@ -151,16 +122,81 @@ function registerDocumentUpdateHandlers(languageService: ILanguageService) {
   return subscriptions;
 }
 
-/**
- * Loads the Q# compiler including the WASM module
- */
+async function activateLanguageService(extensionUri: vscode.Uri) {
+  const subscriptions: vscode.Disposable[] = [];
+
+  const languageService = await loadLanguageService(extensionUri);
+
+  // synchronize document contents
+  subscriptions.push(...registerDocumentUpdateHandlers(languageService));
+
+  // synchronize configuration
+  subscriptions.push(registerConfigurationChangeHandlers(languageService));
+
+  // diagnostics
+  subscriptions.push(...startCheckingQSharp(languageService));
+
+  // completions
+  subscriptions.push(
+    vscode.languages.registerCompletionItemProvider(
+      qsharpDocumentFilter,
+      createCompletionItemProvider(languageService),
+      "."
+    )
+  );
+
+  // hover
+  subscriptions.push(
+    vscode.languages.registerHoverProvider(
+      qsharpDocumentFilter,
+      createHoverProvider(languageService)
+    )
+  );
+
+  // go to def
+  subscriptions.push(
+    vscode.languages.registerDefinitionProvider(
+      qsharpDocumentFilter,
+      createDefinitionProvider(languageService)
+    )
+  );
+
+  return subscriptions;
+}
+
 async function loadLanguageService(baseUri: vscode.Uri) {
   const wasmUri = vscode.Uri.joinPath(baseUri, "./wasm/qsc_wasm_bg.wasm");
   const wasmBytes = await vscode.workspace.fs.readFile(wasmUri);
   await loadWasmModule(wasmBytes);
   return await getLanguageService();
 }
 
+function registerConfigurationChangeHandlers(
+  languageService: ILanguageService
+) {
+  return vscode.workspace.onDidChangeConfiguration((event) => {
+    if (event.affectsConfiguration("qsharp.targetProfile")) {
+      let targetProfile = vscode.workspace
+        .getConfiguration("qsharp")
+        .get<string>("targetProfile");
+      log.info(`Target profile changed: ${targetProfile}`);
+
+      switch (targetProfile) {
+        case "base":
+        case "full":
+          break;
+        default:
+          log.warn(`Invalid value for target profile: ${targetProfile}`);
```

Comment:
They may well have removed the setting, which is totally valid and shouldn't be a warning, so may want to not warn on 'undefined'